finite-state/shared/src/main/scala/fs2/data/esp/Pattern.scala

```diff
@@ -25,153 +25,142 @@ import cats.syntax.all._
 /** A pattern to be matched.
   * A pattern can capture a submatch, that is then usable in the RHS.
   */
-sealed trait Pattern[+Tag] {
-  def |[Tag1 >: Tag](that: Pattern[Tag1]): Pattern[Tag1] =
+sealed trait Pattern[Guard, Tag] {
```

Comment:
Any specific reason you removed the covariance on `Tag`?

Reply:
To make it compile with scala 2.12 and scala 3, for which type inference in patterns for types with 2 covariant parameters seems broken.